pgmpy/estimators/PC.py

```diff
@@ -174,24 +183,38 @@ def estimate(
                 "For using Chi Square or Pearsonr, data argument must be specified"
             )
 
+        if expert_knowledge is None:
+            expert_knowledge = ExpertKnowledge()
+
         # Step 1: Run the PC algorithm to build the skeleton and get the separating sets.
         skel, separating_sets = self.build_skeleton(
             ci_test=ci_test,
-            max_cond_vars=max_cond_vars,
             significance_level=significance_level,
             variant=variant,
             n_jobs=n_jobs,
+            expert_knowledge=expert_knowledge,
+            enforce_background_knowledge=enforce_background_knowledge,
             show_progress=show_progress,
             **kwargs,
         )
 
         if return_type.lower() == "skeleton":
             return skel, separating_sets
 
-        # Step 2: Orient the edges based on build the PDAG/CPDAG.
-        pdag = self.skeleton_to_pdag(skel, separating_sets)
+        # Step 2: Orient the edges based on Meek's rules to build the PDAG/CPDAG.
+        pdag = self.orient_colliders(skel, separating_sets)
+        pdag = self.apply_orientation_rules(pdag)
+
+        # Step 3: Either return the CPDAG, integrate expert knowledge or fully orient the edges to build a DAG.
+        if (
+            expert_knowledge.required_edges != set()
+            or expert_knowledge.forbidden_edges != set()
+        ) and enforce_background_knowledge is False:
+            pdag = expert_knowledge.check_against_pdag(pdag)
+            pdag = self.apply_orientation_rules(pdag, apply_r4=True)
+
+            return pdag
```

Comment:
I don't think the `return` statement should be here? The following post-processing still needs to be done before returning.